### PR TITLE
feat(core): Upgrade FxBaseRootTunnel contracts to be compatible with Polygon events emitted by EIP-1559 transactions

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: nicholaspai <npai.nyc@gmail.com>

**Motivation**

The FxBaseRootTunnel base contract was changed [in this commit](https://github.com/fx-portal/contracts/commit/a3637b4eca9ef5019d4336f7af6b3b20953c52ee) to handle transactions sent with `type = 1` on Polygon for which we want to [generate burn proofs to send messages from Polygon to mainnet](https://docs.polygon.technology/docs/develop/l1-l2-communication/state-transfer#state-transfer-from-polygon-to-ethereum). We will need to redeploy the OracleChild/Root tunnel system after merging this PR.
- [New FX base root tunnel contract](https://github.com/fx-portal/contracts/blob/baed24d22178201bca33140c303e0925661ec0ac/contracts/tunnel/FxBaseRootTunnel.sol#L103)

Additionally, the `maticjs` SDK has breaking changes so this PR also changes the `fx-tunnel-relayer` bot accordingly. 
- [docs for the maticjs library that generates burn proofs](https://maticnetwork.github.io/matic.js/docs/advanced/exit-util/)

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts/external/polygon/lib/ExitPayloadReader.sol [hunk 1]
```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.8.0;
```
Comment (mrice32) [inline packages/core/contracts/external/polygon/lib/ExitPayloadReader.sol:1]: Can you paste the source of this code in these contracts?

File: packages/core/contracts/external/polygon/lib/ExitPayloadReader.sol [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
```
Comment (mrice32) [inline packages/core/contracts/external/polygon/lib/ExitPayloadReader.sol:2]: I'm assuming all contract changes are pure copy-paste? Or did you make modifications?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/core/contracts/external/polygon/lib/ExitPayloadReader.sol (hunk 2 above)
Comment (nicholaspai) [reply packages/core/contracts/external/polygon/lib/ExitPayloadReader.sol:2]: Pure copy-paste, check the change log

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/fx-tunnel-relayer/README.md [hunk 3]
```diff
@@ -10,26 +10,45 @@ This bot is specific to the Polygon-Ethereum communication layer whose architect
...
+Polygon-Ethereum communication differs based on the direction that a message is sent. If a message is sent from Ethereum to Polygon, then the Polygon [State Sync](https://docs.polygon.technology/docs/contribute/state-sync/state-sync/) mechanism takes over. This relies on Polygon validators to detect `StateSynced` events emitted by the Ethereum [StateSender](https://docs.polygon.technology/docs/contribute/state-sync/how-state-sync-works) contract. Validators are incentivized to pick up these events and submit corresponding metadata to a receiver contract on the Polygon network. The metadata includes a target contract and ABI data that the receiver contract can use to forward a smart contract call. Therefore the Ethereum-to-Polygon messaging is handled automatically by Polygon validators.
```
Comment (chrismaree) [inline packages/fx-tunnel-relayer/README.md:13]: thank you for updating these and being so detail focused!

**Review by chrismaree — APPROVED** [review]
good fix! thank you for updating all the associated docs!

**Comment by nicholaspai** [comment]
@chrismaree @mrice32 great call to import from an NPM package. I'm able to remove the entire `external/polygon` folder and use imported contracts instead! I ran diffs manually on the contracts and they look exactly like the ones previously copied into this PR

**Review by chrismaree — APPROVED** [review]

**Pull request merged**